[BLOG] Add joining_labs (#610)
Quansight/Quansight-website: apps/labs/posts/joining_labs.md

---
title: 'Thoughts on joining Quansight Labs'
published: April 8, 2019
author: ralf-gommers

> description: ''

Reviewer: ```suggestion
description: 'Ralf Gommers's perspectives for pushing forward the Python ecosystem for scientific computing and data science, and how to fund it.'
```

Reviewer: @rgommers Is this a good description here? If not, maybe you can suggest an alternative?

Reviewer 2: How about: Ralf Gommers's thoughts on joining Quansight Labs as its director, and his perspectives for pushing forward the Python ecosystem for scientific computing and data science, and how to fund it.

Reviewer: Thanks!

> description: 'Ralf Gommers's thoughts on joining Quansight Labs as its director, and his perspectives for pushing forward the Python ecosystem for scientific computing and data science, and how to fund it.'

Reviewer: ```suggestion
description: "Ralf Gommers's thoughts on joining Quansight Labs as its director, and his perspectives for pushing forward the Python ecosystem for scientific computing and data science, and how to fund it."
```

category: [Community, PyData Ecosystem]
featuredImage:
  src: /posts/joining_labs/community_landscape.png
  alt: 'Image of a word cloud emphasizing the interrelationship of Projects with Assets, Funding and Community.'
hero:
  imageSrc: /posts/joining_labs/blog_hero_var1.svg
  imageAlt: 'An illustration of a brown hand holding up a microphone, with some graphical elements highlighting the top of the microphone.'
---

> In his [blog post](https://www.quansight.com/post/welcoming-ralf-gommers-as-director-of-quansight-labs) welcoming me, Travis set out his vision for pushing forward the Python ecosystem for scientific computing and data science, and how to fund it. In this post I'll add my own perspectives to that. Given that Quansight Labs' purpose, it seems fitting to start with how I see things as a community member and organizer.

Reviewer: @bskinn this link is broken because of the Quansight blog migration. Would it be the same link after you add this back? If not, could you please correct this, so it will work once you migrate the linked blog post?

Reviewer 3: This link will not need to change. We're migrating all the posts so that they live at the same URLs on the new site as the old.

If you find a link that worked on the old site but doesn't on the new, let me know ASAP. That means either:

1. It's a post that hasn't been migrated yet, has a known link, and should be prioritized; or
2. It's a post that *has* been migrated, but we made a mistake with the URL.

### A Quansight Labs perspective

To date, the work at Quansight Labs has mainly focused on innovation and highly experimental prototypes.
Travis already provided some context on [XND](https://xnd.io/) (a clean, modular implementation of generalized NumPy concepts) and [uarray](https://github.com/Quansight-Labs/uarray) (a Python array interface). XND is already fairly mature while uarray is still very young; however both are at the stage where we can think about adoption pathways. Experimental prototypes with a lot of potential include [python-moa](https://github.com/Quansight-Labs/python-moa) (a Python implementation of Mathematics of Arrays), [metadsl](https://github.com/Quansight-Labs/metadsl) (a framework for creating domain specific language libraries in Python), and [mtypes](https://github.com/Quansight-Labs/mtypes) ("memory types for Python", bridging the Python type system to and from C).

The balance of work at Quansight Labs will gradually shift towards sustaining core projects in the PyData ecosystem.

> Which projects we work on will be driven by both availability of funding and personal interests of people. As well as by input from the community - we want to push forward core projects, so we need to act on priorities that core development teams have. Clearly articulating those priorities isn't always simple. Half a year ago at the NumFOCUS Summit I took stock of the roadmaps for all NumFOCUS projects: 50% of the projects didn't have a roadmap, and of the 50% that did again half said that their roadmaps were not up to date. This is also the kind of thing Quansight Labs can help with. For example, Anthony Scopatz has done a great job in producing roadmap brochures for each of the projects that have been covered in the [Open Source Directions](https://www.quansight.com/open-source-directions) webinars: https://www.quansight.com/projects. In that process he helped improve those project roadmaps tremendously.

Reviewer: @bskinn These 2 links are broken due to the website redesign. Is there a good replacement here?
CC @rgommers

Reviewer 2: https://www.youtube.com/playlist?list=PLA7hukZmf3AqDMFZD1Oi5cOoe2svop979 for OSD.

https://www.quansight.com/projects can just be removed.

Reviewer 3: @rgommers Are those roadmap brochures archived somewhere we could reference? I know they're out of date, but they might be an interesting historical artifact.

Reviewer 2: Only on GDrive: https://drive.google.com/drive/folders/11prFjz1ZvhBKAM_z8g82lTyNbkjWAE9e. They were floating around as pdf's in Wix I think.

Reviewer 2: I wouldn't bother spending time on that anymore if it requires republishing them I think.

Reviewer 3: We could share the GDrive folder publicly view-only, I guess... That's kind of brittle/smelly/unprofessional, though.

We _can_ serve arbitrary files from Storyblok's assets feature.... We could zip them all into a single archive and link to that?

Reviewer 2: Serving one zipfile of pdf's is fine with me, if that's not much work. Sharing GDrive isn't a good idea indeed.

Reviewer 3: Or, serve the ZIP in the same way we serve blog images, static site data.

Reviewer 3: I assume what needs to be shared are the 22 GSlides in the subfolders you linked to?

Matter of a few minutes, I think, to download each as a PDF, zip, and upload to SB or commit to repo.

Reviewer 2: Not all were published, some are works in progress. There are some that say "This is a community-led project, ...." on the last page, those are mostly fine. Some others say "LEARN MORE sales@quansight.com" - those are no good.

Reviewer 3: Looking at this again, I realize that the roadmaps weren't published as part of this post. So, publishing them now would be a novel effort, and can be delayed for later or skipped entirely. Whatever the eventual choice, it shouldn't block migration of this post. Concur that the projects link can just be removed.

> Which projects we work on will be driven by both availability of funding and personal interests of people. As well as by input from the community - we want to push forward core projects, so we need to act on priorities that core development teams have. Clearly articulating those priorities isn't always simple. Half a year ago at the NumFOCUS Summit I took stock of the roadmaps for all NumFOCUS projects: 50% of the projects didn't have a roadmap, and of the 50% that did again half said that their roadmaps were not up to date. This is also the kind of thing Quansight Labs can help with. For example, Anthony Scopatz has done a great job in producing roadmap brochures for each of the projects that have been covered in the [Open Source Directions](https://www.quansight.com/open-source-directions) webinars: https://www.quansight.com/projects. In that process he helped improve those project roadmaps tremendously.

Reviewer: ```suggestion
Which projects we work on will be driven by both availability of funding and personal interests of people. As well as by input from the community - we want to push forward core projects, so we need to act on priorities that core development teams have. Clearly articulating those priorities isn't always simple. Half a year ago at the NumFOCUS Summit I took stock of the roadmaps for all NumFOCUS projects: 50% of the projects didn't have a roadmap, and of the 50% that did again half said that their roadmaps were not up to date. This is also the kind of thing Quansight Labs can help with. For example, Anthony Scopatz has done a great job in producing roadmap brochures for each of the projects that have been covered in the [Open Source Directions](https://www.youtube.com/playlist?list=PLA7hukZmf3AqDMFZD1Oi5cOoe2svop979) webinars. In that process he helped improve those project roadmaps tremendously.
```
